test_conformance/basic/test_global_linear_id.cpp

```diff
@@ -22,105 +22,93 @@
 {
-    cl_mem streams;
-    cl_program program[2];
-    cl_kernel kernel[2];
+    clProgramWrapper program[2];
+    clKernelWrapper kernel[2];
 
-    int *output_ptr;
-    size_t threads[2];
+    const char *kernel_names[] = { "test_global_linear_id_1d",
+                                   "test_global_linear_id_2d" };
+    const char *kernel_code[] = { global_linear_id_1d_code,
+                                  global_linear_id_2d_code };
     int err;
-    num_elements = (int)sqrt((float)num_elements);
-    int length = num_elements * num_elements;
-
-    output_ptr = (int *)malloc(sizeof(int) * length);
-
-    streams = clCreateBuffer(context, CL_MEM_READ_WRITE, length * sizeof(int),
-                             NULL, &err);
-    test_error( err, "clCreateBuffer failed.");
-
-    err = create_single_kernel_helper(context, &program[0], &kernel[0], 1,
-                                      &global_linear_id_1d_code,
-                                      "test_global_linear_id_1d");
-    test_error( err, "create_single_kernel_helper failed");
-    err = create_single_kernel_helper(context, &program[1], &kernel[1], 1,
-                                      &global_linear_id_2d_code,
-                                      "test_global_linear_id_2d");
-    test_error( err, "create_single_kernel_helper failed");
-
-    err  = clSetKernelArg(kernel[0], 0, sizeof streams, &streams);
-    test_error( err, "clSetKernelArgs failed.");
-    err  = clSetKernelArg(kernel[1], 0, sizeof streams, &streams);
-    test_error( err, "clSetKernelArgs failed.");
-
-    threads[0] = (size_t)num_elements;
-    threads[1] = (size_t)num_elements;
-    err = clEnqueueNDRangeKernel(queue, kernel[1], 2, NULL, threads, NULL, 0, NULL, NULL);
-    test_error( err, "clEnqueueNDRangeKernel failed.");
-
-    err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0, length*sizeof(int), output_ptr, 0, NULL, NULL);
-    test_error( err, "clEnqueueReadBuffer failed.");
-
-    err = verify_global_linear_id(output_ptr, length);
-
-    threads[0] = (size_t)num_elements;
-    err = clEnqueueNDRangeKernel(queue, kernel[1], 1, NULL, threads, NULL, 0, NULL, NULL);
-    test_error( err, "clEnqueueNDRangeKernel failed.");
-
-    err = clEnqueueReadBuffer(queue, streams, CL_TRUE, 0, num_elements*sizeof(int), output_ptr, 0, NULL, NULL);
-    test_error( err, "clEnqueueReadBuffer failed.");
-
-    err = verify_global_linear_id(output_ptr, num_elements);
-
-    // cleanup
-    clReleaseMemObject(streams);
-    clReleaseKernel(kernel[0]);
-    clReleaseKernel(kernel[1]);
-    clReleaseProgram(program[0]);
-    clReleaseProgram(program[1]);
-    free(output_ptr);
+
+    num_elements = static_cast<int>(sqrt(static_cast<float>(num_elements)));
+    int length = 1;
+    size_t threads[] = { static_cast<size_t>(num_elements),
+                         static_cast<size_t>(num_elements) };
+
+    for (int i = 0; i < ARRAY_SIZE(program) && !err; i++)
```

Comment:
we are checking `&& !err` here, but `int err;` on line 77 isn't getting initialized before the first loop iteration. Think line 77 should be `int err = CL_SUCCESS;`

Reply:
yeap you are right, thanks for spotting that. I'll fix asap